src/app/participant-list/participant-list.component.ts

```diff
@@ -1895,9 +1899,33 @@ export class ParticipantListComponent implements OnInit {
         break;
       }
     }
+    if (!result && participant.data && participant.data.activities && participant.data.activities) {
+      let setting = this.findSettingByColumnName(name);
+      this.participantPage = new ParticipantPageComponent(this.auth, this.compService, this.dsmService, this.router, this.role, this.util, this.route, null);
```

Comment:
I am afraid that creating object of component manually should be a good idea, why was it necessary? It will be created automatically by angular once parent component will be created.